### PR TITLE
Fix/reset kyc

### PR DESCRIPTION
**Pull request by TheLeoB**
## Description (optional)
For security, we are now resetting a user's kyc to Tier 0 when recovering.

The tokens/user IDs needed to call the reset API are available if calling /users with the JWT token, so I moved this call out of the loginRoutineSaga to earlier, when we first restore wallet information from metadata from mnemonic. 

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: .well-known/apple-app-site-association [hunk 1]
```diff
@@ -1,30 +1 @@
...
-        ]
-      }
-    ]
-  },
-  "webcredentials": {
-    "apps": [ "NZ6PH75U7K.com.rainydayapps.Blockchain" ]
-  }
-}
\ No newline at end of file
+{"applinks":{"apps":[],"details":[{"appID":"NZ6PH75U7K.com.rainydayapps.Blockchain","paths":["NOT /_/*","/*"]}]}}
```
Comment (blockdylanb) [inline .well-known/apple-app-site-association:1]: This looks scary, was this intended?

File: packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts [hunk 2]
```diff
@@ -512,7 +512,27 @@ export default ({ api, coreSagas }) => {
         coreSagas.wallet.restoreWalletCredentialsFromMetadata,
         mnemonic
       )
-      yield put(actions.auth.restoreFromMetadataSuccess(metadataInfo))
+      const { guid, sharedKey } = metadataInfo
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts:515]: Could you possibly add some comments on this block to help me understand what's going on here?

File: packages/blockchain-wallet-v4-frontend/src/scenes/RecoverWallet/SecondStep/error.template.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  flex-direction: column;
+`
+
+const Error = (props) => {
+  return (
+    <Wrapper>
+      <Body>
+        <Icon color='error' name='close-circle' size='40px' />
+        <Text size='20px' weight={600} color='black' style={{ marginTop: '8px' }}>
+          Restoring Account Failed
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/scenes/RecoverWallet/SecondStep/error.template.js:21]: FormattedMessage?

**Review by blockdylanb — APPROVED** [review]
Just a few small things, but the code looks good.

File: .well-known/apple-app-site-association (hunk 1 above)
Comment (milan-bc) [reply .well-known/apple-app-site-association:1]: also file missing empty line at the end

**Review by milan-bc — COMMENTED** [review]
In general looks good, there are some comments from Dylan which should be addressed

File: .well-known/apple-app-site-association (hunk 1 above)
Comment (TheLeoB) [reply .well-known/apple-app-site-association:1]: Yep, this is the ios mobile site association file, so smart links will open the ios app. iOS team asked me to update it. 

Line added.

File: packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts (hunk 2 above)
Comment (TheLeoB) [reply packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts:515]: Added!

**Pull request merged**